classes/ReassuranceActivity.php

```diff
@@ -0,0 +1,189 @@
+            'link' => array('type' => self::TYPE_STRING, 'shop' => true, 'lang' => true, 'validate' => 'isUrl', 'required' => false, 'size' => 255),
+            'date_add' => array('type' => self::TYPE_DATE, 'shop' => true, 'validate' => 'isDate'),
+            'date_upd' => array('type' => self::TYPE_DATE, 'shop' => true, 'validate' => 'isDate'),
+        )
+    );
+
+    /**
+     * @param int $id_lang
+     * @param int $id_shop
+     *
+     * @return array
+     */
+    public static function getAllBlockByLang($id_lang = 1, $id_shop = 1)
+    {
+        $sql = 'SELECT * FROM `' . _DB_PREFIX_ . 'psreassurance` pr
+            LEFT JOIN ' . _DB_PREFIX_ . 'psreassurance_lang prl ON (pr.id_psreassurance = prl.id_psreassurance)
+            WHERE prl.id_lang = "' . (int)$id_lang . '" AND prl.id_shop = "' . (int)$id_shop . '"
+            ORDER BY pr.position';
+
+        $result = Db::getInstance()->executeS($sql);
+
+        return $result;
+    }
+
+    /**
+     * @param array $psr_languages
+     * @param int $type_link
+     */
+    public function handleBlockValues($psr_languages, $type_link)
+    {
+        $languages = Language::getLanguages();
+        $newValues = [];
+
+        foreach ($psr_languages as $key => $value) {
+            $newValues[$key] = [
+                'title' => $value->title,
+                'description' => $value->description,
+                'url' => $value->url,
+            ];
+        }
+
+        foreach ($languages as $language) {
+            if (false === array_key_exists($language['id_lang'], $newValues)) {
+                continue;
+            }
+
+            $this->title[$language['id_lang']] = $newValues[$language['id_lang']]['title'];
+            $this->description[$language['id_lang']] = $newValues[$language['id_lang']]['description'];
+
+            if ($type_link === self::TYPE_LINK_URL) {
+                $this->link[$language['id_lang']] = $newValues[$language['id_lang']]['url'];
+            } else {
+                $this->link[$language['id_lang']] = '';
+            }
+        }
+
+        // @todo: where does $id_cms comes from ?
+        if (isset($id_cms) && $type_link === self::TYPE_LINK_CMS_PAGE) {
+            $this->id_cms = $id_cms;
+            $link = Context::getContext()->link;
+
+            foreach ($languages as $language) {
+                $this->link[$language['id_lang']] = $link->getCMSLink(
+                    $id_cms,
+                    null,
+                    null,
+                    $language['id_lang']
+                );
+            }
+        }
+
+        if ($type_link == 'undefined') {
+            $type_link = self::TYPE_LINK_NONE;
+        }
+
+        $this->type_link = $type_link;
+    }
+
+    /**
+     * @param int $id_shop
+     *
+     * @return array
+     */
+    public static function getAllBlockByShop($id_shop = 1)
```

Comment:
You cannot be sure the default value would be one. If you except to load data from the default shop, you should consider `Configuration::get('PS_SHOP_DEFAULT')`